### ISSUE TITLE
TypeError: Cannot read property 'longname' of null

### ISSUE THREAD

**Issue by alexisab** [jsdoc3/jsdoc#1416]
I try to setup jsdoc in my project. I've write some simple comments but when I run jsdoc i got this error : 
```
/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/parser.js:360
        result.memberof = doclet.longname + jsdoc.name.SCOPE.PUNC.INSTANCE;
                                ^

TypeError: Cannot read property 'longname' of null
    at exports.Parser.Parser.astnodeToMemberof (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/parser.js:360:33)
    at addSymbolMemberof (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/handlers.js:252:31)
    at newSymbolDoclet (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/handlers.js:282:13)
    at exports.Parser.<anonymous> (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/handlers.js:349:13)
    at emitTwo (events.js:106:13)
    at exports.Parser.emit (events.js:192:7)
    at exports.Visitor.Visitor.visitNode (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/visitor.js:566:16)
    at exports.Visitor.Visitor.visit (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/visitor.js:468:27)
    at exports.Walker.Walker.recurse (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/walker.js:699:38)
    at exports.Parser.Parser._walkAst (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/parser.js:265:18)
    at exports.Parser.Parser._parseSourceCode (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/parser.js:256:18)
    at exports.Parser.Parser.parse (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/lib/jsdoc/src/parser.js:179:18)
    at Object.module.exports.cli.parseFiles (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/cli.js:370:42)
    at module.exports.cli.main (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/cli.js:239:14)
    at Object.module.exports.cli.runCommand (/home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/cli.js:189:5)
    at /home/alexis/Documents/programmation/clovis/client/node_modules/jsdoc/jsdoc.js:105:9
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

Here is my config file : 
```
{
    "plugins": [],
    "recurseDepth": 3,
    "source": {
        "include": [ "src" ],
        "includePattern": ".+\\.js(doc|x)?$",
        "excludePattern": "(^|\\/|\\\\)_"
    },
    "sourceType": "module",
    "tags": {
        "allowUnknownTags": true,
        "dictionaries": ["jsdoc","closure"]
    },
    "templates": {
        "cleverLinks": false,
        "monospaceLinks": false
    },
    "opts": {
        "template": "node_modules/docdash",
        "destination": "./doc/",
        "readme": "./README.md",
        "verbose": true,
        "recurse": true
    }
}
```

I run this command (in my package.json, but I got the same error if I use directly jsdoc bin): 
```
jsdoc -c .jsdoc.json
```

If I remove the `recurse` option it works :confused: 

I'm on ubuntu 16.04,
Node v7.6.0
Yarn 0.27.5
jsdoc 3.5.3

**Comment by hegemonic** [comment]
I can't diagnose this issue without an example of source code that triggers the error. Can you please provide an example?

**Comment by alexisab** [comment]
I've investigate a little bit more and I've found that the problem is from a specific file. I copy paste the code below : 

```

import { connect } from 'react-redux'

import actions from '../actions'
import selectors from '../selectors'
import { TextInput } from '../text-input'
import { ValidationResult } from '../text-input/validation'
import { validatePassword, validateNewPassword } from '../util/password'
import { ScrollPane } from '../scroll-pane'
import { FullscreenPopup } from './popup'

export const EditPassword = connect(state => {
    const me = selectors.me(state)
    const { passwordUpdate } = state

    return {
        me,
        passwordUpdate,
    }
})(class extends React.Component {
    state = {
        old: null,
        new0: null,
        new1: null,
        pendingRequest: false,
        error: null,
    }

    inputs = {}

    send = () => {
        if (!this.valid) {
            return
        }

        const { old, new0 } = this.state
        this.props.dispatch(actions.users.me.password.put.request({
            oldPassword: old,
            newPassword: new0,
        }))
        this.setState({pendingRequest: true})
    }

    componentWillReceiveProps(nextProps) {
        if (!this.state.pendingRequest) {
            return
        }

        const { passwordUpdate } = nextProps
        const { error } = passwordUpdate
        if (error) {
            this.setState({
                error,
                pendingRequest: false,
            })
            return
        }
        if (!this.state.pending) {
            nextProps.onClose()
        }
    }

    validateNew1 = password => {
        if (validateNewPassword(password)) {
            return validateNewPassword(password)
        }

        if (password !== this.state.new0) {
            return new ValidationResult(
                'error', 'Les mots de passe ne correspondent pas')
        }
    }

    get valid() {
        const { old, new0, new1 } = this.state
        return !validatePassword(old) &&
               !validateNewPassword(new0) &&
               !this.validateNew1(new1)
    }

    get buttonDisabled() {
        return this.state.pendingRequest || !this.valid
    }

    get oldError() {
        const { error } = this.state
        if (!error) {
            return null
        }

        if (error.body && error.body.message === 'badOldPassword') {
            return new ValidationResult(
                'error', 'Ancien mot de passe invalide')
        }

        return new ValidationResult(
            'error', 'Une erreur s’est produite')
    }

    oldPasswordChanged = old => {
        this.setState({
            old,
            error: null,
        })
    }

    renderForm = () => (
        <form action="javascript:void(0)">
            <h1>Changement de mot de passe</h1>


            <label>Ancien mot de passe :</label>

            <TextInput placeholder="Ancien mot de passe"
                       type="password"
                       onChange={this.oldPasswordChanged}
                       validator={validatePassword}
                       autoFocus
                       validationResult={this.oldError}
                       ref={input => this.inputs.old = input} />

            <label>Nouveau mot de passe :</label>

            <TextInput placeholder="Nouveau mot de passe"
                       type="password"
                       onChange={new0 => this.setState({new0})}
                       validator={validateNewPassword} />

            <label>Confirmez votre nouveau mot de passe :</label>

            <TextInput placeholder="Confirmez votre nouveau mot de passe"
                       type="password"
                       onChange={new1 => this.setState({new1})}
                       validator={this.validateNew1} />

            <button className="button--primary"
                    type="submit"
                    disabled={this.buttonDisabled}
                    onClick={this.send}>
                Valider
            </button>
        </form>
    )

    render = () => (
        <FullscreenPopup onClose={this.props.onClose}>
            <ScrollPane>
                {this.renderForm()}
            </ScrollPane>
        </FullscreenPopup>
    )
})
```

**Comment by hegemonic** [comment]
Thanks for the example, @alexisab. I can trigger the error with your example and this reduced test case:

```js
export const Foo = connect(state => {
})(class {
    render = () => (
        <FullscreenPopup>
        </FullscreenPopup>
    )
});
```

Investigating.

**Comment by hegemonic** [comment]
Fixed on master and releases/3.5. The fix will be included in JSDoc 3.5.4.
